libcore/ConnectServerDialog.vala

```diff
@@ -0,0 +1,530 @@
+
+        server_entry.focus_out_event.connect (() => {
+            connect_button.sensitive = server_entry.text != "";
+            return false;
+        });
+
+        var port_label = new Gtk.Label (_("Port:"));
+        port_label.xalign = 1;
+        port_spinbutton = new Gtk.SpinButton.with_range (0, ushort.MAX, 1);
+        port_spinbutton.digits = 0;
+        port_spinbutton.numeric = true;
+        port_spinbutton.update_policy = Gtk.SpinButtonUpdatePolicy.IF_VALID;
+
+        var type_label = new Gtk.Label (_("Type:"));
+        type_label.xalign = 1;
+        var type_store = new Gtk.ListStore (2, typeof (MethodInfo), typeof (string));
+        type_combobox = new Gtk.ComboBox.with_model (type_store);
+        var renderer = new Gtk.CellRendererText ();
+        type_combobox.pack_start (renderer, true);
+        type_combobox.add_attribute (renderer, "text", 1);
+
+        var share_label = new Gtk.Label (_("Share:"));
+        share_label.xalign = 1;
+        share_entry = new Gtk.Entry ();
+        share_entry.bind_property ("visible", share_label, "visible", GLib.BindingFlags.DEFAULT);
+
+        var folder_label = new Gtk.Label (_("Folder:"));
+        folder_label.xalign = 1;
+        folder_entry = new Gtk.Entry ();
+        folder_entry.text = "/";
+
+        user_header_label = new Granite.HeaderLabel (_("User Details"));
+
+        var domain_label = new Gtk.Label (_("Domain name:"));
+        domain_label.xalign = 1;
+        domain_entry = new Gtk.Entry ();
+        domain_entry.bind_property ("visible", domain_label, "visible", GLib.BindingFlags.DEFAULT);
+
+        var user_label = new Gtk.Label (_("User name:"));
+        user_label.xalign = 1;
+        user_entry = new Gtk.Entry ();
+        user_entry.bind_property ("visible", user_label, "visible", GLib.BindingFlags.DEFAULT);
+
+        var password_label = new Gtk.Label (_("Password:"));
+        password_label.xalign = 1;
+        password_entry = new Gtk.Entry ();
+        password_entry.visibility = false;
+        password_entry.bind_property ("visible", password_label, "visible", GLib.BindingFlags.DEFAULT);
+
+        remember_checkbutton = new Gtk.CheckButton.with_label (_("Remember this password"));
+        password_entry.bind_property ("visible", remember_checkbutton, "visible", GLib.BindingFlags.DEFAULT);
+
+        cancel_button = new Gtk.Button.with_label (_("Cancel"));
+        cancel_button.show ();
+        cancel_button.clicked.connect (on_cancel_clicked);
+
+        connect_button = new Gtk.Button.with_label (_("Connect"));
+        connect_button.no_show_all = true;
+        connect_button.show ();
+        connect_button.clicked.connect (on_connect_clicked);
+
+        continue_button = new Gtk.Button.with_label (_("Continue"));
+        continue_button.no_show_all = true;
+        continue_button.hide ();
+        continue_button.clicked.connect (on_continue_clicked);
+
+        ((Gtk.Container)(get_action_area ())).add (cancel_button);
```

Comment:
To avoid deprecation warnings about `get_action_area ()` you can do something like `continue_button = add_button (_("Continue"));`

Reply:
That was the in the first version but there are problems with that - using that function associates the button with a response signal and I do not want the buttons always to emit a response because that causes the dialog to close prematurely in some cases.  Although this is a subclass of Gtk.Dialog it continues to be active after the "Connect button" is pressed so that it can deal with requests from the mount operation to enter further details.  Also during the "connect phase" pressing "Cancel" should end the mount operation but not necessarily close the dialog.

Maybe I can put the buttons in the content area instead.